---
title: "Analysis COVID-19"
author: "Diego Iván Morales Gallardo"
date: "2023-07-05"
output: html_document
---

### Virus List
NS-1  
Sino1-11  
GZ02  
GD01  
Wuhan-Hu-1  
2019-nCoV/USA-WA1/2020  
Tor2  
MA15  
TW1  
ZS-C  
LC5  
LC3  
ZS-B  
HSZ-Cc  
HZS2-C  
HSZ2-A  
GZ-C  
SZ16  
TW9  
HGZ8L1-A  
SZ3  

```{r}
# Vector with virus GenBank accession codes
virus <- c("JX869059", "AY508724", "MN908947", "AY390556", "MN985325", "NC_004718", "DQ497008", 
           "AY291451", "AY395003", "AY395002", "AY395000", "AY394996", "AY394995", "AY394992", 
           "AY394983", "AY394979", "AY304488", "AY502932", "AY394981", "AY304486")
```

```{r}
# Install necessary packages if not already installed
if (!require("BiocManager", quietly = TRUE))
  install.packages("BiocManager")
BiocManager::install("Biostrings")
BiocManager::install("ggtree")
BiocManager::install("DECIPHER")
BiocManager::install("ggmsa")
install.packages("adegenet")
install.packages("seqinr")

# Load required libraries
library(seqinr)
library(ape)
library(phytools)
library(Biostrings)
library(ggtree)
library(DECIPHER)
library(viridis)
library(ggplot2)
library(adegenet)
```

```{r}
# Load virus sequences from GenBank
virus_sequences <- read.GenBank(virus)
sample_seq <- read.GenBank("JX869059")
str(virus_sequences)
attributes(virus_sequences)
names(virus_sequences)
attr(virus_sequences, "species")

# Write sequences to a FASTA file
write.dna(virus_sequences, file = "virus_seqs.fasta", format = "fasta", append = FALSE, nbcol = 6, colsep = " ", colw = 10)

# Read sequences from FASTA
virus_seq_unaligned <- readDNAStringSet("virus_seqs.fasta", format = "fasta")
virus_seq_unaligned <- OrientNucleotides(virus_seq_unaligned)
virus_align_seqs <- AlignSeqs(virus_seq_unaligned)

# Save aligned sequences to a new FASTA file
writeXStringSet(virus_align_seqs, file = "virus_align_seq.fasta")
virus_aligned <- read.alignment("virus_align_seq.fasta", format = "fasta") 
```

```{r}
# Print sequence lengths
print("Lengths of sequences:")
for (i in 1:20) {
  print(length(virus_sequences[[i]]))
}
```

```{r}
# Read aligned sequences from FASTA for further analysis
SARS_COV2 <- read.fasta("virus_seqs.fasta")

# Function to compare base frequencies
compare_sequences <- function(seq1) {
  par(mfrow = c(1, 4))
  for (i in 1:20) {
    barplot(table(seq1[i]), col = 1:5)
  }
}
compare_sequences(SARS_COV2)
```

```{r}
# Generate a phylogenetic tree
distance_matrix <- dist.alignment(virus_aligned, matrix = "similarity")
phylo_tree <- nj(distance_matrix)
ladderized_tree <- ladderize(phylo_tree)
phylo_plot <- ggtree(ladderized_tree) + geom_tiplab()
phylo_plot
```

### Analysis and Interpretation

The graphs generated by the code present a phylogenetic analysis of 20 different viruses, including SARS-CoV-2 and other human coronaviruses responsible for similar illnesses.

In the first graph, which shows nucleotide base comparisons in virus sequences, we observe the frequency of each base at each sequence position. This allows us to identify genetic similarities and differences among the analyzed viruses. For SARS-CoV-2 and other human coronaviruses, we can look for common patterns in nucleotide bases that may be associated with their ability to infect humans.

The second graph displays the phylogenetic tree built from the aligned sequences. This tree shows the evolutionary relationships among the viruses, where closer branches indicate greater genetic similarity. We can observe the position of SARS-CoV-2 in the tree and determine which viruses are most closely related.

Based on these analyses, we can conclude the following:

- SARS-CoV-2 shares genetic similarities with other human coronaviruses causing similar diseases, such as MERS. This supports the hypothesis that these viruses belong to the same family and share a common ancestor.
- By analyzing nucleotide bases, it is possible to identify specific patterns in SARS-CoV-2 and other human coronavirus sequences. These patterns could be associated with the functions and characteristics of these viruses, such as their ability to infect humans.
- The phylogenetic tree shows the evolutionary relationships among the viruses. We can identify which viruses are most closely related to SARS-CoV-2 and assess their degree of genetic similarity.

### References:

- COVID-19 Map - Johns Hopkins Coronavirus Resource Center. (n.d.). Johns Hopkins Coronavirus Resource Center. [https://coronavirus.jhu.edu/map.html](https://coronavirus.jhu.edu/map.html)
- SARS-CoV-2 Variant Tracking. (n.d.). [https://www.who.int/en/activities/tracking-SARS-CoV-2-variants](https://www.who.int/en/activities/tracking-SARS-CoV-2-variants)
- Official Sources for COVID-19 Information in Ibero-American Countries. Fundación Gabo. (2019). [https://fundaciongabo.org/en/news/article/official-sources-to-consult-information-about-covid-19-in-iberoamerican-countries](https://fundaciongabo.org/en/news/article/official-sources-to-consult-information-about-covid-19-in-iberoamerican-countries)
- Zhou, P., Yang, X. L., Wang, X. G., Hu, B., Zhang, L., Zhang, W., ... & Shi, Z. L. (2020). A pneumonia outbreak associated with a new coronavirus of probable bat origin. *Nature*, 579(7798), 270-273.
- Lam, T. T., Shum, M. H., Zhu, H. C., Tong, Y. G., Ni, X. B., Liao, Y. S., ... & Cheung, W. Y. (2020). Identifying SARS-CoV-2 related coronaviruses in Malayan pangolins. *Nature*, 583(7815), 282-285.
